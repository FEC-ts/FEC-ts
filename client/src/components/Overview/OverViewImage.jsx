import react,  {useState, useEffect, useRef}from 'react';

export default function OverViewImage({currentStyle}){

  const [currentImage, setCurrentImage] = useState('');

  useEffect(()=> {
    if(currentStyle.style_id !== undefined){

      setCurrentImage(currentStyle.photos[0].url);
    }

  }, [currentStyle]);


  if(currentStyle.style_id  !== undefined){
    return (<div className="currentImageDiv">
      <img src={currentImage}></img>
    </div>)
  }

  return (
    <progress></progress>
  )
}