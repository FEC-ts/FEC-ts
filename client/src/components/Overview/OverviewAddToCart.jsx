import react,  {useState, useEffect, useRef}from 'react';
import axios from 'axios';
export default function({currentStyle}){
  const [sizeObj, setSizeObj] = useState({});
  const [sizeArray, setSizeArray] =useState([]);
  const quantityRef = useRef();
  const sizeRef = useRef();
  useEffect(() => {

    if(currentStyle.style_id !== undefined) {
      setSizeObj({});
      var tempObj = {};
      setSizeArray([]);
      if(sizeRef.current !== undefined) {
        sizeRef.current.value = 'Select Size';
        setSelectedSize('Select Size');
      }
      for (var i in currentStyle.skus) {

        if (currentStyle.skus[i].quantity >= 1) {
          tempObj[currentStyle.skus[i].size] = {};
          tempObj[currentStyle.skus[i].size].quantity = currentStyle.skus[i].quantity;
          tempObj[currentStyle.skus[i].size].sku_id = i;

          let temp = currentStyle.skus[i].size;

          setSizeArray(sizeArray => [...sizeArray, temp]);

        }
      }
      setSizeObj(tempObj);


    }

  }, [currentStyle])



  const [selectedSize, setSelectedSize] = useState();

  const sizeSelectOnChange = (e) => {
    setSelectedSize(e.target.value);
    messageRef.current.style.visibility = 'hidden';
    sizeRef.current.size = 1;
  }

  const [selectedSizeQuantity, setSelectedSizeQuantity] = useState(['-'])
  useEffect(()=> {
    if(selectedSize === 'Select Size') {
      quantityRef.current.disabled =true;
      setSelectedSizeQuantity(['-']);


      return;
    }

    if(Object.keys(sizeObj).length > 0) {
      quantityRef.current.disabled = false;


      if (sizeObj[selectedSize].quantity >= 15) {

        setSelectedSizeQuantity([]);
        for (var i = 1; i < 16; i++) {
          let temp = i;
          setSelectedSizeQuantity(selectedSizeQuantity => [...selectedSizeQuantity, temp]);
        }
      } else {
        setSelectedSizeQuantity([]);
        for (var i = 1; i <= sizeObj[selectedSize].quantity; i++) {
          let temp = i;
          setSelectedSizeQuantity(selectedSizeQuantity => [...selectedSizeQuantity, temp]);
        }
      }
      quantityRef.current.value = 1;
    }

  }, [selectedSize])

  const messageRef = useRef();





  const addToCartOnClick = (e) => {
    if (sizeRef.current.value === 'Select Size') {

      sizeRef.current.size = 5;
      messageRef.current.style.visibility = 'visible';
    } else{
      let q = quantityRef.current.value;
      let id = sizeObj[selectedSize].sku_id;
      let temp = {sku_id: id};
      let url = '/cart';

      for (var i = 0; i < q; i++) {
        axios.post(url, temp)
          .then(result => console.log(result.data))
          .catch(err => console.log(err));
      }

    }
    axios.get('/cart')
    .then(result => console.log(result.data))
          .catch(err => console.log(err));


  };


  if (Object.keys(sizeObj).length > 0) {

    return (
      <>
       <div className="selectSizeMessageDiv" ref={messageRef} style={{visibility:"hidden"}}><p>Please select size</p></div>
      <div className="sizeDiv">
      <select className="sizeSelection" onChange={sizeSelectOnChange} ref={sizeRef}>
        <option   value="Select Size">Select Size</option>
        {
          sizeArray.map(i => {

            return (
              <option value={i}>{i}</option>
            )
          })
        }
      </select>

    </div>

    <div className="quantityDiv">
      <select className="quantitySelection"  ref={quantityRef} disabled>
        {selectedSizeQuantity.map(e => {

          return (
            <option value={e}>{e}</option>
          )
        })}
      </select>
    </div>

    <div className="addToCartbuttonDiv">
      <input type='button' onClick={addToCartOnClick} value='Add To Cart'></input>
    </div>

      </>
    )

  }

  return ( <>

    <div className="sizeDiv">
      <select className="sizeSelection" disabled>
        <option value="OUT OF STOCK" >OUT OF STOCK</option>
      </select>

    </div>

    <div className="quantityDiv">
      <select className="quantitySelection" disabled>
        <option value="-" >-</option>
      </select>
    </div>


  </>



  )


}