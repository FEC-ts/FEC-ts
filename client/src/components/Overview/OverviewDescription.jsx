import react,  {useState, useEffect, useRef}from 'react';
import axios from 'axios';
import {Facebook, Twitter, Pinterest} from './../icons/OverviewShareIconsSVG.jsx'

export default function({itemInfo}) {
  const [slogan, setSlogan] = useState(itemInfo.slogan);
  const [description, setDescription] = useState(itemInfo.description);
  const [features, setFeatures] = useState(itemInfo.features)

  useEffect(()=> {
    setSlogan(itemInfo.slogan);
    setDescription(itemInfo.description);
    setFeatures(itemInfo.features);
  }, [itemInfo])

  const facbookCurrentURL = `http://www.facebook.com/sharer.php?u=${window.location.href}`
  const twitterCurrentURL = `https://twitter.com/intent/tweet?url='${window.location.href}'`;
  const pinterestCurrentURL = `http://pinterest.com/pin/create/button/?url=${window.location.href}`

  return (
    <>
      <div className="sloganDiv">
        <b>{slogan}</b>
      </div>

      <div className="descriptionDiv">
        <p>{description}</p>
      </div>

      <div className="shareDiv" >

        <div className="facebookShareDiv" >
          <a href={facbookCurrentURL} target="_blank"><Facebook/></a>
        </div>

        <div className="twitterShareDiv"  >
          <a href={twitterCurrentURL} target="_blank"><Twitter/></a>
        </div>

        <div className="pinterestShareDiv">
        <a href={pinterestCurrentURL} target="_blank"><Pinterest/></a>
        </div>
      </div>

      <div className="featuresDiv">
        {features.map(e => {
          return (
            <p>{e.feature}: <span>{e.value}</span></p>

          )
        })}
      </div>

    </>
  )
}