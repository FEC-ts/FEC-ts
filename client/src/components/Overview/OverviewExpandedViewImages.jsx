import react,  {useState, useEffect, useRef}from 'react';

export default function ({e, index, eRef, setProductImagesButtonDisplay, imageIndex,productImages, setimgRefState}){
  const imgRef = useRef();
  const imgDivRef = useRef();
  const [zoomIn, setZoomIn] = useState(false);

  const imageDivOnClick = () => {
    setZoomIn(!zoomIn);
  }

  useEffect(()=> {
    if (zoomIn) {
      imgRef.current.style = "object-fit:fill; height: 225vh; width: 225vw;"
      setProductImagesButtonDisplay(0);
    } else {
      imgRef.current.style = "object-fit:contain; height: 90vh; width: 90vw;"
      if(imageIndex === 0){
        setProductImagesButtonDisplay(2);
      } else if(imageIndex === productImages.length-1){
        setProductImagesButtonDisplay(1);
      } else {
        setProductImagesButtonDisplay(3);
      }
    }
  },[zoomIn])

  const imgOnMouseMove = (e)=>{

    if (zoomIn) {
      imgRef.current.style.cursor = "zoom-out";
      let x = e.clientX - eRef.current.offsetLeft;
      let y = e.clientY - eRef.current.offsetTop -5;
      x *= 1.5;
      y *= 1.5;

      imgRef.current.style.transform = `translate(${-x}px, -${y}px)`
      setimgRefState([imgRef, setZoomIn, zoomIn]);

    }

  }

  return (
    <div className="expandCurrentImagesDiv" onMouseMove={imgOnMouseMove} ref={imgDivRef}>
      <img className="expandCurrentImage"   onClick={imageDivOnClick}  src={e} ref={imgRef}></img>
    </div>
  )
}