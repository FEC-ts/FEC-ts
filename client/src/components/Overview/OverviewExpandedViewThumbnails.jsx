import react,  {useState, useEffect, useRef, useContext}from 'react';
import {InteractionAPIContext} from './../InteractionAPI.jsx';
export default function({e, indexe, index, setImageIndex, imageIndex, prevImageIndex, setPrevImageIndex, setProductImageThumbnailsYIndex, currentStyle, productImageThumbnailsYIndex, productImageThumbnails, imgRefState}){
  const interactionAPI = useContext(InteractionAPIContext);
  const thumbnailRef = useRef();
  useEffect(()=> {
    if (thumbnailRef) {

      if (prevImageIndex === index) {
        thumbnailRef.current.style.border = "1px solid #ddd";
      }
      if (imageIndex === index) {
        thumbnailRef.current.style.border = "3px solid grey";
      }
    }

  }, [imageIndex])

  useEffect(()=> {
    setPrevImageIndex(-1);
    thumbnailRef.current.style.border = "1px solid #ddd";
    setImageIndex(0);
    if(index=== 0){
      thumbnailRef.current.style.border = "3px solid grey";
    }
  }, [currentStyle])

  const thumbnailOnClick = ()=>{
    interactionAPI("expanded view thumbnails", "overview");
    setPrevImageIndex(imageIndex);
    setImageIndex(index);
    setProductImageThumbnailsYIndex(-111 * (index-2) );
    imgRefState[0].current.style = "object-fit:contain; height: 90vh; width: 90vw;"
    imgRefState[1](!imgRefState[2]);
  }


  return (
    <div className="expandCurrentThumbnailsDiv hoverPointer" ref={thumbnailRef} onClick={thumbnailOnClick}>
      <img className="expandCurrentThumbnails" src={e}></img>
    </div>
  )
}