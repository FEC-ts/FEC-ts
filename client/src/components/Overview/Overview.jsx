import react,  {useState, useEffect, useRef, useLayoutEffect}from 'react';
import axios from 'axios';
import OverViewImage from './OverViewImage.jsx';
import OverviewProductInformation from './OverviewProductInformation.jsx';
import OverviewDescription from './OverviewDescription.jsx'
// Group Effort

const Overview = ({productInfo}) => {
  const [itemInfo, setItemInfo] = useState(productInfo);
  const [itemRating, setItemRating] = useState(0);
  const [productStyles, setProductStyles] = useState([]);
  const [currentStyle, setCurrentStyle] = useState({});
  const itemId = productInfo.id;
  const [itemReviews, setItemReviews] = useState();


  useEffect(()=>{
    setProductStyles([]);
    let url = '/products/' + itemId + '/styles';

    axios.get(url)
      .then(result => {

        setProductStyles(productStyles=> [...productStyles, result.data.results]);

          for (var style of result.data.results) {
            if (style['default?']) {

              setCurrentStyle(style);
              break;
            }
          }


      } )
      .catch(err => console.log(err));

      url = '/reviews';
      var params = {params:{product_id: itemId, count: 99999}};
      axios.get(url, params)
      .then(result => {

        setItemReviews(result.data.results.length);
      })
      .catch(err=> console.log(err));

      url = '/reviews/meta';
      params = {params:{product_id: itemId}}
      axios.get(url, params)
      .then(result => {

        var avgRating = 0;
        var count = 0;
        for(var i in result.data.ratings) {
          avgRating += i * result.data.ratings[i];
          count += parseInt(result.data.ratings[i]);
        }

        avgRating /= count;

        setItemRating(avgRating);
      } )
      .catch(err => console.log(err));


  },[productInfo]);


if(currentStyle === undefined) {
  return (
    <progress></progress>
  )
}
  return(
    <div className="currentStyleDiv">
      <h1>Overview</h1>

      <div className="overviewImageDiv">
        <OverViewImage currentStyle={currentStyle}/>
      </div>

      <div className="overviewProductInformationDiv">
          <OverviewProductInformation itemReviews={itemReviews} itemRating={itemRating} productStyles={productStyles} currentStyle={currentStyle} setCurrentStyle={setCurrentStyle}  itemInfo={itemInfo} setItemInfo={setItemInfo}/>
      </div>

      <div className="overviewDescriptionDiv">
        <OverviewDescription itemInfo={itemInfo} ></OverviewDescription>
      </div>



    </div>
  )
};

export default Overview;