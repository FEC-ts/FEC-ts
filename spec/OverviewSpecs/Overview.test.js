/** @jest-environment jsdom */
import renderer from 'react-test-renderer';
import Overview from '../../client/src/components/Overview/Overview.jsx';
import {cleanup, fireEvent, render} from '@testing-library/react';

import react,  {useState, useEffect, useRef, useLayoutEffect,  useContext}from 'react';
const stubs = require('./stubs.js');

var productInfo = stubs.productInfo;
describe('Overview', ()=>{


  it('should render overview section', ()=> {
    const component = renderer.create(<Overview productInfo={productInfo}/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })



})
