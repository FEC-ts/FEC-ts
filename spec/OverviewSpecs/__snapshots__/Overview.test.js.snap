// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should render overview section 1`] = `
<div
  className="currentStyleDiv"
>
  <h1>
    Heir Force Ones
  </h1>
  <div
    className="overviewImageDiv"
  >
    <div
      id="topPage"
      style={
        {
          "display": "none",
        }
      }
    >
      <div
        className="expandViewModal"
      >
        <div
          className="expandCurrentImagesCarouselDiv"
        >
          <div
            className="overviewLeftButtonDiv hoverPointer"
            onClick={[Function]}
            style={
              {
                "display": "inherit",
              }
            }
          >
            <svg
              className="overviewLeftButton"
              viewBox="0 0 8 8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m4 0-4 4 4 4 1.5-1.5-2.5-2.5 2.5-2.5z"
                transform="translate(1)"
              />
            </svg>
          </div>
          <div
            className="overviewRightButtonDiv hoverPointer"
            onClick={[Function]}
            style={
              {
                "display": "inherit",
              }
            }
          >
            <svg
              className="overviewRightButton"
              viewBox="0 0 8 8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m1.5 0-1.5 1.5 2.5 2.5-2.5 2.5 1.5 1.5 4-4z"
                transform="translate(1)"
              />
            </svg>
          </div>
          <div
            className="overviewDismissButtonDiv hoverPointer"
            onClick={[Function]}
          >
            <svg
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m1.89705176 2.05378835.07261815-.08411844c.26626657-.26626656.68293025-.29047261.97654174-.07261815l.08411844.07261815 2.96966991 2.96933009 2.96966991-2.96933009c.29289322-.29289321.76776696-.29289321 1.06066019 0 .2928932.29289322.2928932.76776696 0 1.06066018l-2.9693301 2.96966991 2.9693301 2.96966991c.2662665.26626657.2904726.68293025.0726181.97654174l-.0726181.08411845c-.26626658.2662665-.68293026.2904726-.97654175.0726181l-.08411844-.0726181-2.96966991-2.9693301-2.96966991 2.9693301c-.29289322.2928932-.76776696.2928932-1.06066018 0-.29289321-.29289323-.29289321-.76776697 0-1.06066019l2.96933009-2.96966991-2.96933009-2.96966991c-.26626656-.26626657-.29047261-.68293025-.07261815-.97654174l.07261815-.08411844z"
                fill="#212121"
              />
            </svg>
          </div>
          <div
            className="expandCurrentImagesCarousel"
          />
        </div>
        <div
          className="expandCurrentImageThumbnailsCarouselDiv"
        >
          <div
            className="expandCurrentImageThumbnailsCarousel"
          />
        </div>
      </div>
    </div>
    <div
      className="currentImageThumbnailsCarouselDiv"
    >
      <div
        className="overviewUpButtonDiv hoverPointer"
        onClick={[Function]}
        style={
          {
            "display": "inherit",
          }
        }
      >
        <svg
          className="overviewUpButton"
          fill="none"
          viewBox="0 0 12 12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2.14645 7.35355c.19526.19527.51184.19527.7071 0l3.14645-3.14644 3.14645 3.14644c.19526.19527.51184.19527.7071 0 .19525-.19526.19525-.51184 0-.7071l-3.5-3.5c-.19526-.19527-.51184-.19527-.7071 0l-3.5 3.5c-.19527.19526-.19527.51184 0 .7071z"
            fill="#212121"
          />
        </svg>
      </div>
      <div
        className="currentImageThumbnailsCarousel"
      />
      <div
        className="overviewDownButtonDiv hoverPointer"
        onClick={[Function]}
        style={
          {
            "display": "inherit",
          }
        }
      >
        <svg
          className="overviewDownButton"
          fill="none"
          viewBox="0 0 12 12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m2.14645 4.64645c.19526-.19527.51184-.19527.7071 0l3.14645 3.14644 3.14645-3.14644c.19526-.19527.51184-.19527.7071 0 .19525.19526.19525.51184 0 .7071l-3.5 3.5c-.19526.19527-.51184.19527-.7071 0l-3.5-3.5c-.19527-.19526-.19527-.51184 0-.7071z"
            fill="#212121"
          />
        </svg>
      </div>
    </div>
    <div
      className="currentImagesCarouselDiv"
    >
      <div
        className="OverviewButtons"
      >
        <div
          className="overviewLeftButtonDiv hoverPointer"
          onClick={[Function]}
          style={
            {
              "display": "inherit",
            }
          }
        >
          <svg
            className="overviewLeftButton"
            viewBox="0 0 8 8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m4 0-4 4 4 4 1.5-1.5-2.5-2.5 2.5-2.5z"
              transform="translate(1)"
            />
          </svg>
        </div>
        <div
          className="overviewRightButtonDiv hoverPointer"
          onClick={[Function]}
          style={
            {
              "display": "inherit",
            }
          }
        >
          <svg
            className="overviewRightButton"
            viewBox="0 0 8 8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1.5 0-1.5 1.5 2.5 2.5-2.5 2.5 1.5 1.5 4-4z"
              transform="translate(1)"
            />
          </svg>
        </div>
        <div
          className="expandViewIconDiv hoverPointer"
          onClick={[Function]}
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m7.5955055 15.3496699c.29289322-.2928932.76776695-.2928932 1.06066017 0s.29289322.767767 0 1.0606602l-3.09516567 3.0938844h2.189c.37969577 0 .69349096.2821539.74315338.6482294l.00684662.1017706c0 .4142136-.33578644.75-.75.75h-4c-.41421356 0-.75-.3357864-.75-.75v-4c0-.4142136.33578644-.75.75-.75s.75.3357864.75.75v2.189zm8.6595052 5.6545446c-.4142136 0-.75-.3357864-.75-.75s.3357864-.75.75-.75h2.186l-3.0915492-3.0940929c-.2661619-.2663712-.2902041-.6830444-.0722343-.9765702l.0726513-.0840899c.2930083-.2927781.767882-.2925914 1.06066.0004169l3.0944722 3.0963361v-2.192c0-.3796958.2821539-.693491.6482294-.7431534l.1017706-.0068466c.4142136 0 .75.3357864.75.75v4c0 .4142136-.3357864.75-.75.75zm-8.5050107-18.0042145c.41421356 0 .75.33578644.75.75s-.33578644.75-.75.75h-2.187l3.09253851 3.09487842c.26616188.26637121.29020414.68304438.07223428.97657021l-.07265121.08408988c-.29300833.29277806-.76788203.2925914-1.06066009-.00041693l-3.09446149-3.09712158v2.192c0 .37969577-.28215388.69349096-.64822944.74315338l-.10177056.00684662c-.41421356 0-.75-.33578644-.75-.75v-4c0-.41421356.33578644-.75.75-.75zm12.5049977 0c.4142136 0 .75.33578644.75.75v4c0 .41421356-.3357864.75-.75.75s-.75-.33578644-.75-.75v-2.191l-3.0945604 3.09622284c-.2662127.26632039-.6828715.2906107-.976527.07281562l-.0841331-.07260115c-.2929525-.29283398-.2930485-.76770771-.0002145-1.06066015l3.092435-3.09477716h-2.187c-.3796958 0-.693491-.28215388-.7431534-.64822944l-.0068466-.10177056c0-.41421356.3357864-.75.75-.75z"
              fill="#212121"
            />
          </svg>
        </div>
      </div>
      <div
        className="currentImagesCarousel"
      />
    </div>
  </div>
  <div
    className="overviewProductInformationDiv"
  >
    <div
      className="overviewRating"
    >
      <svg
        className="OverviewRatingStars"
        fill="none"
        viewBox="0 0 12 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
          fill="#212121"
        />
      </svg>
      <svg
        className="OverviewRatingStars"
        fill="none"
        viewBox="0 0 12 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
          fill="#212121"
        />
      </svg>
      <svg
        className="OverviewRatingStars"
        fill="none"
        viewBox="0 0 12 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
          fill="#212121"
        />
      </svg>
      <svg
        className="OverviewRatingStars"
        fill="none"
        viewBox="0 0 12 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
          fill="#212121"
        />
      </svg>
      <svg
        className="OverviewRatingStars"
        fill="none"
        viewBox="0 0 12 12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
          fill="#212121"
        />
      </svg>
      <span
        className="relatedItemRatingTooltip"
      >
         
        <a
          href="#ratingsReviews"
          onClick={[Function]}
        >
          <b>
            Read all 
             reviews
          </b>
        </a>
      </span>
    </div>
    <div
      className="overviewCategoryDiv"
    >
      <p
        className="overviewCategory"
      >
         Category:  
        <code>
          Kicks
        </code>
      </p>
    </div>
    <div
      className="overviewNameDiv"
    >
      <p
        className="overviewName"
      >
         Product:  
        <code>
          Heir Force Ones
        </code>
      </p>
    </div>
    <div
      className="overviewPriceDiv"
    >
      <p
        className="overviewPrice"
      >
         Price:  $
      </p>
    </div>
    <div
      className="overviewStylesDiv"
    >
      <p
        className="overviewCurrentStyleName"
      >
        <span>
          STYLE  &gt;   
        </span>
        <code />
      </p>
      <progress />
    </div>
    <div
      className="overviewAddToCartDiv"
    >
      <div
        className="sizeDiv"
      >
        <select
          className="sizeSelection"
          disabled={true}
        >
          <option
            value="OUT OF STOCK"
          >
            OUT OF STOCK
          </option>
        </select>
      </div>
      <div
        className="quantityDiv"
      >
        <select
          className="quantitySelection"
          disabled={true}
        >
          <option
            value="-"
          >
            -
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="overviewDescriptionDiv"
  >
    <div
      className="sloganDiv"
    >
      <b>
        A sneaker dynasty
      </b>
    </div>
    <div
      className="descriptionDiv"
    >
      <p>
        Now where da boxes where I keep mine? You should peep mine, maybe once or twice but never three times. I'm just a sneaker pro, I love Pumas and shell toes, but can't nothin compare to a fresh crispy white pearl
      </p>
    </div>
    <div
      className="shareDiv"
    >
      <b>
        Share:   
      </b>
      <div
        className="facebookShareDiv"
      >
        <a
          href="http://www.facebook.com/sharer.php?u=http://localhost/"
          onClick={[Function]}
          target="_blank"
        >
          <svg
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className="shareIcons"
              d="m22 16.19c0 3.64-2.17 5.81-5.81 5.81h-1.19c-.55 0-1-.45-1-1v-5.77c0-.27.22-.5.49-.5l1.76-.03c.14-.01.26-.11.29-.25l.35-1.91c.03-.18-.11-.35-.3-.35l-2.13.03c-.28 0-.5-.22-.51-.49l-.04-2.45c0-.16.13-.29999.3-.29999l2.4-.04001c.17 0 .3-.12999.3-.29999l-.04-2.40002c0-.17-.13-.29999-.3-.29999l-2.7.04001c-1.66.03-2.98 1.38999-2.95 3.04999l.05 2.75c.01.28-.21.5-.49.51l-1.2.02c-.17 0-.29999.13-.29999.3l.03 1.9c0 .17.12999.3.29999.3l1.2-.02c.28 0 .5.22.51.49l.09 5.7c.01.56-.44 1.02-1 1.02h-2.3c-3.64 0-5.81-2.17-5.81-5.82v-8.37c0-3.64 2.17-5.81 5.81-5.81h8.38c3.64 0 5.81 2.17 5.81 5.81z"
              fill="#613B6F"
            />
          </svg>
        </a>
      </div>
       
      <div
        className="twitterShareDiv"
      >
        <a
          href="https://twitter.com/intent/tweet?url='http://localhost/'"
          onClick={[Function]}
          target="_blank"
        >
          <svg
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className="shareIcons"
              d="m31.937 6.093c-1.177.516-2.437.871-3.765 1.032 1.355-.813 2.391-2.099 2.885-3.631-1.271.74-2.677 1.276-4.172 1.579-1.192-1.276-2.896-2.079-4.787-2.079-3.625 0-6.563 2.937-6.563 6.557 0 .521.063 1.021.172 1.495-5.453-.255-10.287-2.875-13.52-6.833-.568.964-.891 2.084-.891 3.303 0 2.281 1.161 4.281 2.916 5.457-1.073-.031-2.083-.328-2.968-.817v.079c0 3.181 2.26 5.833 5.26 6.437-.547.145-1.131.229-1.724.229-.421 0-.823-.041-1.224-.115.844 2.604 3.26 4.5 6.14 4.557-2.239 1.755-5.077 2.801-8.135 2.801-.521 0-1.041-.025-1.563-.088 2.917 1.86 6.36 2.948 10.079 2.948 12.067 0 18.661-9.995 18.661-18.651 0-.276 0-.557-.021-.839 1.287-.917 2.401-2.079 3.281-3.396z"
              fill="#613B6F"
            />
          </svg>
        </a>
      </div>
       
      <div
        className="pinterestShareDiv"
      >
        <a
          href="http://pinterest.com/pin/create/button/?url=http://localhost/"
          onClick={[Function]}
          target="_blank"
        >
          <svg
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              className="shareIcons"
              d="m16.021 0c-8.828 0-15.984 7.156-15.984 15.984 0 6.771 4.214 12.552 10.161 14.88-.141-1.266-.266-3.203.052-4.583.292-1.25 1.875-7.943 1.875-7.943s-.479-.964-.479-2.375c0-2.219 1.292-3.88 2.891-3.88 1.365 0 2.026 1.021 2.026 2.25 0 1.37-.87 3.422-1.323 5.323-.38 1.589.797 2.885 2.365 2.885 2.839 0 5.026-2.995 5.026-7.318 0-3.813-2.75-6.49-6.677-6.49-4.547 0-7.214 3.417-7.214 6.932 0 1.375.526 2.854 1.188 3.651.13.161.146.302.109.464-.12.5-.391 1.599-.443 1.818-.073.297-.229.359-.536.219-1.99-.922-3.245-3.839-3.245-6.193 0-5.036 3.667-9.672 10.563-9.672 5.542 0 9.854 3.958 9.854 9.229 0 5.516-3.474 9.953-8.307 9.953-1.62 0-3.141-.839-3.677-1.839l-1 3.797c-.359 1.391-1.339 3.135-2 4.193 1.5.458 3.078.714 4.734.714 8.813 0 15.979-7.151 15.979-15.984 0-8.828-7.167-15.979-15.979-15.979z"
              fill="#613B6F"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="featuresDiv"
    >
      <p>
        <b>
          Sole
          : 
        </b>
        <span>
          Rubber
        </span>
      </p>
      <p>
        <b>
          Material
          : 
        </b>
        <span>
          FullControlSkin
        </span>
      </p>
      <p>
        <b>
          Mid-Sole
          : 
        </b>
        <span>
          ControlSupport Arch Bridge
        </span>
      </p>
      <p>
        <b>
          Stitching
          : 
        </b>
        <span>
          Double Stitch
        </span>
      </p>
    </div>
  </div>
</div>
`;
